File: spring-cloud-integrated-demo/order-service/src/main/scala/com/github/jccode/springcloud/integrateddemo/order/form/OrderItemForm.scala
package com.github.jccode.springcloud.integrateddemo.order.form

import javax.validation.constraints.{Min, NotNull}

import scala.beans.BeanProperty

class OrderItemForm {

  @BeanProperty @NotNull @Min(1) var productId: Integer = _
  @BeanProperty @NotNull @Min(1) var amount: Integer = _

  override def toString = s"OrderItemForm($productId, $amount)"
}

File: spring-cloud-integrated-demo/order-service/src/main/java/com/github/jccode/springcloud/integrateddemo/order2/form/OrderItemForm.java
package com.github.jccode.springcloud.integrateddemo.order2.form;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;

public class OrderItemForm {

    @NotNull
    @Min(1)
    private int productId;

    @NotNull
    @Min(1)
    private int amount;

    public int getProductId() {
        return productId;
    }

    public void setProductId(int productId) {
        this.productId = productId;
    }

    public int getAmount() {
        return amount;
    }

    public void setAmount(int amount) {
        this.amount = amount;
    }
}
